vscode/src/workspace.ts

```diff
@@ -342,26 +361,14 @@ export class Workspace implements WorkspaceInterface {
     return result;
   }
 
-  private registerRestarts(context: vscode.ExtensionContext) {
-    this.createRestartWatcher(context, "Gemfile.lock");
-    this.createRestartWatcher(context, "gems.locked");
-    this.createRestartWatcher(context, "**/.rubocop.yml");
-    this.createRestartWatcher(context, ".rubocop");
+  private registerRestarts() {
+    this.createRestartWatcher(this.context, `{${WATCHED_FILES.join(",")}}`);
 
     // If a configuration that affects the Ruby LSP has changed, update the client options using the latest
     // configuration and restart the server
-    context.subscriptions.push(
+    this.context.subscriptions.push(
       vscode.workspace.onDidChangeConfiguration(async (event) => {
         if (event.affectsConfiguration("rubyLsp")) {
-          // Re-activate Ruby if the version manager changed
-          if (
-            event.affectsConfiguration("rubyLsp.rubyVersionManager") ||
-            event.affectsConfiguration("rubyLsp.bundleGemfile") ||
-            event.affectsConfiguration("rubyLsp.customRubyCommand")
-          ) {
-            await this.ruby.activateRuby();
-          }
```

Comment:
This already happens on `start`, so this code was actually causing duplicate Ruby env activation.